relay-server/src/actors/project_local.rs

```diff
@@ -136,31 +144,42 @@ fn load_local_states(projects_path: &Path) -> io::Result<HashMap<ProjectKey, Arc
     Ok(states)
 }
 
-fn poll_local_states(
-    manager: Addr<LocalProjectSource>,
-    config: Arc<Config>,
-) -> impl Future<Item = (), Error = ()> {
-    let (sender, receiver) = oneshot::channel();
+async fn poll_local_states(path: &Path, tx: &mpsc::Sender<HashMap<ProjectKey, Arc<ProjectState>>>) {
+    let states = load_local_states(path).await;
+    match states {
+        Ok(states) => {
+            let res = tx.send(states).await;
+            if res.is_err() {
+                relay_log::error!("failed to store static project configs");
+            }
+        }
+        Err(error) => relay_log::error!(
+            "failed to load static project configs: {}",
+            LogError(&error)
+        ),
+    };
+}
 
-    let _ = thread::spawn(move || {
-        let path = config.project_configs_path();
-        let mut sender = Some(sender);
+async fn spawn_poll_local_states(
```

Comment:
only one thing I this, should this function be moved to the top, before it used first time? 

and same for `poll_local_states`

Reply:
It might make the diff harder to read, but yes, absolutely.